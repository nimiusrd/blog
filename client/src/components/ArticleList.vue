<template>
    <ul class="article--list">
        <ArticleItem
            v-for="item in items"
            :author="item.author"
            :id="item.id"
            :key="item.key"
            :postedAt="item.postedAt"
            :title="item.title"
            :updatedAt="item.updatedAt"
        />
    </ul>
</template>

<script>
import Vue from 'vue'
import ArticleItem from './ArticleItem.vue'

const ArticleList = Vue.extend({
    components: {
        ArticleItem
    },
    data: () => ({
        items: [
            {
                author: 'author1',
                id: 1,
                postedAt: '2019-09-19T13:18:32.000Z',
                title: 'title1',
                updatedAt: '2019-09-19T13:18:32.000Z'
            },
            {
                author: 'author2',
                id: 2,
                postedAt: '2019-09-19T13:18:32.000Z',
                title: 'title2',
                updatedAt: '2019-09-19T13:18:32.000Z'
            },
        ]
    })
})

export default ArticleList
</script>
