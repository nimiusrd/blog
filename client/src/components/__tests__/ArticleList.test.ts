import ArticleList from '@/components/ArticleList.vue'
import {shallowMount} from '@vue/test-utils'

test('snapshot', () => {
    const wrapper = shallowMount(ArticleList, {
        methods: {
            getArticles() {
                this.$data.items = [
                    {
                        author: 'author',
                        content: 'content',
                        postedAt: '2019-09-28T06:34:56+00:00',
                        title: 'title',
                        updatedAt: '2019-09-28T06:34:56+00:00'
                    }
                ]
            }
        }
    })

    expect(wrapper.html()).toMatchSnapshot()
})
