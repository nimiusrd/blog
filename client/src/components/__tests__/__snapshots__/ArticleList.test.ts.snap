// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `"<div class=\\"list\\"><articlelistheader-stub></articlelistheader-stub> <articlelistitem-stub author=\\"author\\" postedat=\\"2019-09-28T06:34:56+00:00\\" title=\\"title\\" updatedat=\\"2019-09-28T06:34:56+00:00\\"></articlelistitem-stub></div>"`;
